### DISCUSSION TITLE
Force Layout Presets

### DISCUSSION THREAD

**Discussion by rasco** [directus/directus#13668] [Feature Requests]
I've had issues with (especially new) users not seeing what they need to see in the collection listings.

The post below by rijk in another issue reminded me about that problem.

I think it would be very useful if I could "force" users to get a certain layout.
I know this is already possible in settings->layouts, but it's quite a lot of work to get it right. And you constantly need to keep up with changes.
If the admin user had a button on each collection that allowed to reset all users' collection layouts to the current one used, that would be very useful. I could easily define sensible defaults and maybe optionally lock users to them.

> Where would those custom names be saved? I suppose it would be on the preset of the layout options, though those are per collection and per role/user, so maybe that's not super in the UX dept. You wouldn't want to use the data-model settings for this either, as you do in fact want to scope it per layout I suppose, as it's only really mostly relevant for fields you nest through a relationship, however, if it's per layout that config isn't shared/reused between different layouts, which would be pretty inconvenient if you switch from like a configured map preview to the tabular layout 🤔

_Originally posted by @rijkvanzanten in https://github.com/directus/directus/discussions/13277#discussioncomment-2858626_

**Comment by azrikahar** [comment]
> I've had issues with (especially new) users not seeing what they need to see in the collection listings.

Would you mind clarifying the issues? Perhaps with something like "what is the current result" vs "what is the expected outcome", and/or screenshots or relevant mockups.

In relation to the requested clarification as I may not fully understand the description here, just to confirm, is this more so a feature request or an actual existing bug?

**Reply by rasco** [reply]
### What is the current result?
A team member registers and starts using our Directus backend. 
When they first open any collections, their tables will have "random" columns, often in nonsensical order.
It's very confusing for a new user obviously.

Right now, the only way to prevent that seems to be in "Presets & Bookmarks" and, I guess, make some global "Scope: All users" table layouts. Although that doesn't affect already existing users. To do it for existing users, I'd have to look for their presets for that collection and delete them, I guess?
Anyway, the Presets & Bookmarks table somehow doesn't show some fields, or has a bad table layout itself. When I try to change it, it always resets back to what it was. That makes it hard to quickly select the rows that you're looking for. See screenshot
<img width="1529" alt="Screenshot 2022-06-02 at 7 48 44 PM" src="https://user-images.githubusercontent.com/289536/171622867-836fccfe-4615-49a2-afd9-dd0b4914a462.png">
 
### What is the expected outcome?
I'd like to be able to easily force table layouts on the users, current and existing users.
Maybe just a button in the collection view sidebar.

Of course it's great that Directus offers so much flexibility with table layouts, but it's also important that all (especially new) users see the system in the same way. One reason is that it makes it easier to communicate about things.

**Reply by benhaynes** [reply]
The default order _should_ be the first X columns in their native order... if not, we should fix that.

What you described is very true, and poor UX. The main thing we need is a button beside the bookmark icon in the header that allows users to _reset_ their preset to the Role/Global configuration (delete the personal preset for that collection). We have bigger plans for overhauling the setup for presets, but this smaller change should be easy and is crucial. CC @directus/team

**Reply by rijkvanzanten** [reply]
> columns in their native order... 

Just FYI, most databases don't actually have a "native order", so this is not something we can rely on 👍🏻

**Reply by benhaynes** [reply]
Sorry, I meant the order we have them in Data Model.

**Comment by piotr-cz** [comment]
For me the expected outcome is to disable creating presets and editing/ deleting existing presets by non-admin users.
Some data types make sense for Table layout, but not for Calendar/ Map.

At this moment every time user changes preset layout or it's options, new preset is being created automatically for him/ her.

Unfortunately it's not possible to lock presets to predefined ones with Roles, as Permissions are set to *Use Custom* (Required for App Access).

**Reply by piotr-cz** [reply]
Now I see that presets are used to hold users state (for example current month in tabular view)

**Comment by goggi** [comment]
One option could be to add a icon close to the Collection name that makes it possible for the users to reset their preset to the global one (if a global preset exists). Only thing that needs to be done really is to delete the user preset for that collection and it will revert to the global one.

**Comment by erondpowell** [comment]
When we resolve this, please tag me for a docs update! 🤠

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
